haas/cli.py

```diff
@@ -130,9 +175,33 @@ def setup_http_client():
         http_client = RequestsHTTPClient()
         http_client.auth = (basic_username, basic_password)
         return
-    # Fall back to no authentication:
-    http_client = RequestsHTTPClient()
-
+    # Next try keystone:
+    try:
+        from keystoneauth1.identity import v3
+        from keystoneauth1 import session
+        os_auth_url = os.getenv('OS_AUTH_URL')
+        os_password = os.getenv('OS_PASSWORD')
+        os_username = os.getenv('OS_USERNAME')
+        os_user_domain_id = os.getenv('OS_USER_DOMAIN_ID') or 'Default'
+        os_project_name = os.getenv('OS_PROJECT_NAME')
+        os_project_domain_id = os.getenv('OS_PROJECT_DOMAIN_ID') or 'Default'
+        if None in (os_auth_url, os_username, os_password, os_project_name):
+            # XXX: This error kinda makes sense, but I (zenhack) am somewhat
+            # dissatisfied.
+            raise KeyError
+        auth = v3.Password(auth_url=os_auth_url,
+                           username=os_username,
+                           password=os_password,
+                           project_name=os_project_name,
+                           user_domain_id=os_user_domain_id,
+                           project_domain_id=os_project_domain_id)
+        sess = session.Session(auth=auth)
+        http_client = KeystoneHTTPClient(sess)
+        return
+    except (ImportError, KeyError):
+        pass
```

Comment:
I don't think we need an explicit variable - we can pick an order (either OpenStack or DB Auth first, the other second, and `Null` third), then check for the accompanying environmental variables. Just need to be sure to document it conspicuously (like in the command help).

BTW - the [keystoneclient documentation](http://docs.openstack.org/developer/python-keystoneclient/using-sessions.html#service-discovery) makes it sound like service discovery should be an easy addition, adding a single argument to the session.get/put/post calls. Of course, looks can be deceiving :-)